### PR TITLE
Add E0452 error explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Manishearth

### CODE REVIEW

File: src/librustc/diagnostics.rs [hunk 1]
```diff
@@ -1943,6 +1943,20 @@ fn main() {
...
 "##,
 
+E0452: r##"
+An invalid lint attribute has been given. Erroneous code example:
+
+```
+#![allow(foo = "")] // error: malformed lint attribute
+```
+
+You need to pass an ident as lint attribute to fix this issue:
```
Comment (Manishearth) [inline src/librustc/diagnostics.rs]: "Lint attributes only accept a list of identifiers (where each identifier is a lint name). Ensure the attribute is of this form:"

**Comment by Manishearth** [comment]
@bors retry

O.o

**Comment by Manishearth** [comment]
```
../src/librustc/middle/expr_use_visitor.rs:285:9: 285:68 error: cannot infer an appropriate lifetime for lifetime parameter `'tcx` due to conflicting requirements
../src/librustc/middle/expr_use_visitor.rs:285         ExprUseVisitor { typer: typer, mc: mc, delegate: delegate }
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/librustc/middle/expr_use_visitor.rs:279:5: 286:6 help: consider using an explicit lifetime parameter as shown: fn new(delegate: &'d mut (Delegate<'tcx>),
       typer: &'t infer::InferCtxt<'a, 'a>) -> ExprUseVisitor<'d, 't, 'a, 'a>
 where 'tcx:'a'd:
../src/librustc/middle/expr_use_visitor.rs:279     pub fn new(delegate: &'d mut (Delegate<'tcx>),
../src/librustc/middle/expr_use_visitor.rs:280                typer: &'t infer::InferCtxt<'a, 'tcx>)
../src/librustc/middle/expr_use_visitor.rs:281                -> ExprUseVisitor<'d,'t,'a,'tcx> where 'tcx:'a+'d
../src/librustc/middle/expr_use_visitor.rs:282     {
../src/librustc/middle/expr_use_visitor.rs:283         let mc: mc::MemCategorizationContext<'t, 'a, 'tcx> =
../src/librustc/middle/expr_use_visitor.rs:284             mc::MemCategorizationContext::new(typer);
                                               ...
error: aborting due to previous error
make: *** [x86_64-apple-darwin/stage0/lib/rustlib/x86_64-apple-darwin/lib/stamp.rustc] Error 101
```

looks like there are some old files left over on the build machine?

**Comment by GuillaumeGomez** [comment]
cc @alexcrichton

**Comment by alexcrichton** [comment]
Unfortunately I'm not entirely sure what's going on here, I've seen this sort of error crop up from time to time but it always seems unrelated to the change at hand (e.g. like this PR). This may be a stage0 bug that has since been fixed (not sure), and it may just involve waiting for some other PRs to land for this one to stop being unlucky. Beyond that I'm not so sure :(

**Comment by nikomatsakis** [comment]
This is due to a bug in the snapshot, I believe. This bug has been fixed. Unfortunately random permutations of the source can lead to this error.

**Comment by nikomatsakis** [comment]
Maybe time for a new snapshot!

**Pull request merged**